---
title: "dmrseq Results Analysis"
author: "Aaron Kovacs & Mark Ziemann"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    fig_width: 7
    fig_height: 5
theme: cosmo
---

This code is available at https://github.com/aaronsk7/guppy-methylation

## dmrs before annotation

```{r, Load in packages}

library(GenomicFeatures)
library(BiocManager)
library(RCircos)
library(dplyr)
library(data.table)
library(kableExtra)
library(knitr)

```


```{r, Load in dmrs}
dmrseqCvG <- readRDS("/mnt/data/aaron/projects/guppy-methylation/dmrseqrds/dmrseqCvG.rds")
dmrseqCvL <- readRDS("/mnt/data/aaron/projects/guppy-methylation/dmrseqrds/dmrseqCvL.rds")
dmrseqGvL <- readRDS("/mnt/data/aaron/projects/guppy-methylation/dmrseqrds/dmrseqGvL.rds")
dmrseqCvG
head(dmrseqCvL)
head(dmrseqGvL)
CvGdmrs <- as.data.frame(dmrseqCvG,row.names = 1:nrow(as.data.frame(ranges(dmrseqCvG))))
CvGdmrs <- dplyr::select(CvGdmrs,seqnames,width,L,beta,pval,qval)
colnames(CvGdmrs) <- c("location","width","L","beta","pvalue","qvalue")
CvGdmrs <- head(CvGdmrs,10)
CvGdmrs
LightComp <- c('CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG')
CvGdmrs <- cbind(LightComp,CvGdmrs)
saveRDS(object=CvGdmrs,file="/mnt/data/aaron/projects/guppy-methylation/Thesis/figures/Tables/CvGpreanodmrs.rds")

CvLdmrs <- as.data.frame(dmrseqCvL,row.names = 1:nrow(as.data.frame(ranges(dmrseqCvL))))
CvLdmrs <- dplyr::select(CvLdmrs,seqnames,width,L,beta,pval,qval)
colnames(CvLdmrs) <- c("location","width","L","beta","pvalue","qvalue")
CvLdmrs <- head(CvLdmrs,10)
LightComp <- c('CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL')
CvLdmrs <- cbind(LightComp,CvLdmrs)
saveRDS(object=CvLdmrs,file="/mnt/data/aaron/projects/guppy-methylation/Thesis/figures/Tables/CvLpreanodmrs.rds")

GvLdmrs <- as.data.frame(dmrseqGvL,row.names = 1:nrow(as.data.frame(ranges(dmrseqGvL))))
GvLdmrs <- dplyr::select(GvLdmrs,seqnames,width,L,beta,pval,qval)
colnames(GvLdmrs) <- c("location","width","L","beta","pvalue","qvalue")
GvLdmrs <-head(GvLdmrs,10)
LightComp <- c('GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL')
GvLdmrs <- cbind(LightComp,GvLdmrs)
saveRDS(object=GvLdmrs,file="/mnt/data/aaron/projects/guppy-methylation/Thesis/figures/Tables/GvLpreanodmrs.rds")

LightComp <- c('CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL')
preanodmrs <- union(CvGdmrs,CvLdmrs)
preanodmrs <- union(preanodmrs,GvLdmrs)
preanodmrs <- cbind(LightComp,preanodmrs)
saveRDS(object=preanodmrs,file="/mnt/data/aaron/projects/guppy-methylation/Thesis/figures/Tables/preanodmrs.rds")

```



```{r, Load in genes promoter regions and CpG islands }

gtf <- rtracklayer::import("/mnt/data/aaron/projects/guppy-methylation/Poecilia_reticulata.Guppy_female_1.0_MT.103.gtf")
head(gtf)
my_genes <- gtf[gtf$type == "gene"]
head(my_genes)
mcols(my_genes) <- mcols(my_genes)[c(5,8,18)]
head(my_genes)

gff <- rtracklayer::import("/mnt/data/aaron/projects/guppy-methylation/Poecilia_reticulata.Guppy_female_1.0_MT.dna_sm.toplevel_out.gff")
head(gff)
my_CpGislands <- gff[gff$type == "sequence_feature"]
head(my_CpGislands)
mcols(my_CpGislands) <- mcols(my_CpGislands)[c(1,5)]
head(my_CpGislands)

my_Promotersdf <- read.table("/mnt/data/aaron/projects/guppy-methylation/Poecilia_reticulata.Guppy_female_1.0_MT.103_tss.bed",header = FALSE,stringsAsFactors=FALSE)
head(my_Promotersdf)
my_Promoters <- makeGRangesFromDataFrame(my_Promotersdf,
                         keep.extra.columns=TRUE,
                         ignore.strand=FALSE,
                         seqinfo=NULL,
                         seqnames.field=c("V1"),
                         start.field="V2",
                         end.field=c("V3"),
                         strand.field="V4",
                         starts.in.df.are.0based=FALSE)

head(my_Promoters)

```

## dmrs annotated to genes

```{r, CvG gene annotations}

ol1 <- findOverlaps(dmrseqCvG,my_genes)
ol1
CvGgenes <- dmrseqCvG[queryHits(ol1)]
mygenesol1 <- my_genes[subjectHits(ol1)]
head(CvGgenes)
mygenesol1

CvGgenesdf <- as.data.frame(CvGgenes,row.names = 1:nrow(as.data.frame(ranges(CvGgenes))))
str(CvGgenesdf)
mygenesoldf1 <- as.data.frame(mygenesol1,row.names = 1:nrow(as.data.frame(ranges(mygenesol1))))
head(mygenesoldf1)
CvGgenesdf <- cbind(CvGgenesdf,mygenesoldf1[c("gene_id","gene_biotype","gene_name")])
CvGgenesdf$seqnames <- as.character(CvGgenesdf$seqnames)
head(CvGgenesdf)
CvGgeneschrdf <- CvGgenesdf[grep("LG",CvGgenesdf$seqnames),]
CvGgeneschrdf <- dplyr::select(CvGgeneschrdf,seqnames,start,end,width,L,beta,stat,pval,qval,gene_id,gene_name)
head(CvGgeneschrdf,10)
rm(CvGgenes)
rm(CvGgenesdf)

```


```{r, CvL gene annotations}

ol2 <- findOverlaps(dmrseqCvL,my_genes)
ol2
CvLgenes <- dmrseqCvL[queryHits(ol2)]
mygenesol2 <- my_genes[subjectHits(ol2)]
head(mygenesol2)
CvLgenesdf <- as.data.frame(CvLgenes,row.names = 1:nrow(as.data.frame(ranges(CvLgenes))))
head(CvLgenesdf)
mygenesoldf2 <- as.data.frame(mygenesol2,row.names = 1:nrow(as.data.frame(ranges(mygenesol2))))
head(mygenesoldf2)
CvLgenesdf <- cbind(CvLgenesdf,mygenesoldf2[c("gene_id","gene_biotype","gene_name")])
CvLgenesdf$seqnames <- as.character(CvLgenesdf$seqnames)
CvLgeneschrdf <- CvLgenesdf[grep("LG",CvLgenesdf$seqnames),]
CvLgeneschrdf <- dplyr::select(CvLgeneschrdf,seqnames,start,end,width,L,beta,stat,pval,qval,gene_id,gene_name)
head(CvLgeneschrdf)
rm(CvLgenes)
rm(CvLgenesdf)

```


```{r, GvL gene annotations}

ol3 <- findOverlaps(dmrseqGvL,my_genes)
ol3
GvLgenes <- dmrseqGvL[queryHits(ol3)]
mygenesol3 <- my_genes[subjectHits(ol3)]
head(mygenesol3)
GvLgenesdf <- as.data.frame(GvLgenes,row.names = 1:nrow(as.data.frame(ranges(GvLgenes))))
head(GvLgenesdf)
mygenesoldf3 <- as.data.frame(mygenesol3,row.names = 1:nrow(as.data.frame(ranges(mygenesol3))))
head(mygenesoldf3)
GvLgenesdf <- cbind(GvLgenesdf,mygenesoldf3[c("gene_id","gene_biotype","gene_name")])
GvLgenesdf$seqnames <- as.character(GvLgenesdf$seqnames)
GvLgeneschrdf <- GvLgenesdf[grep("LG",GvLgenesdf$seqnames),]
GvLgeneschrdf <- dplyr::select(GvLgeneschrdf,seqnames,start,end,width,L,beta,stat,pval,qval,gene_id,gene_name)
head(GvLgeneschrdf)
rm(GvLgenes)
rm(GvLgenesdf)

```

```{r, gene annotations table CvG}

CvGgeneschrdf
CvGgeneschrdf <- head(CvGgeneschrdf,10)
CvGgeneschrdf <- dplyr::select(CvGgeneschrdf,seqnames,width,L,beta,pval,qval,gene_name)
CvGgeneschrdf
colnames(CvGgeneschrdf) <- c("location","width","L", "beta","pvalue","qvalue","gene_name")
LightComp <- c('CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG')
LightComp
CvGgeneanodmrs <- cbind(LightComp,CvGgeneschrdf)
CvGgeneanodmrs
saveRDS(object=CvGgeneanodmrs,file="/mnt/data/aaron/projects/guppy-methylation/Thesis/figures/Tables/CvGgeneanodmrs.rds")

```

```{r, gene annotations table CvL}

CvLgeneschrdf
CvLgeneschrdf <- head(CvLgeneschrdf,10)
CvLgeneschrdf <- dplyr::select(CvLgeneschrdf,seqnames,width,L,beta,pval,qval,gene_name)
CvLgeneschrdf
colnames(CvLgeneschrdf) <- c("location","width","L", "beta","pvalue","qvalue","gene_name")
LightComp <- c('CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL')
LightComp
CvLgeneanodmrs <- cbind(LightComp,CvLgeneschrdf)
CvLgeneanodmrs
saveRDS(object=CvLgeneanodmrs,file="/mnt/data/aaron/projects/guppy-methylation/Thesis/figures/Tables/CvLgeneanodmrs.rds")

```

```{r, gene annotations table GvL}

GvLgeneschrdf
GvLgeneschrdf <- head(GvLgeneschrdf,10)
GvLgeneschrdf <- dplyr::select(GvLgeneschrdf,seqnames,width,L,beta,pval,qval,gene_name)
colnames(GvLgeneschrdf) <- c("location","width","L", "beta","pvalue","qvalue","gene_name")
GvLgeneschrdf
LightComp <- c('GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL')
LightComp
GvLgeneanodmrs <- cbind(LightComp,GvLgeneschrdf)
GvLgeneanodmrs
saveRDS(object=GvLgeneanodmrs,file="/mnt/data/aaron/projects/guppy-methylation/Thesis/figures/Tables/GvLgeneanodmrs.rds")

```

## dmrs annotated to CpG islands

```{r, CvG CpG island annotations}

ol1 <- findOverlaps(dmrseqCvG,my_CpGislands)
ol1
CvGCpGislands <- dmrseqCvG[queryHits(ol1)]
my_CpGislandsol1 <- my_CpGislands[subjectHits(ol1)]
head(CvGCpGislands)
my_CpGislandsol1

CvGCpGislandsdf <- as.data.frame(CvGCpGislands,row.names = 1:nrow(as.data.frame(ranges(CvGCpGislands))))
str(CvGCpGislandsdf)
my_CpGislandsol1 <- as.data.frame(my_CpGislandsol1,row.names = 1:nrow(as.data.frame(ranges(my_CpGislandsol1))))
head(my_CpGislandsol1)
CvGCpGislandsdf <- cbind(CvGCpGislandsdf,my_CpGislandsol1[c("source","ID")])
CvGCpGislandsdf$seqnames <- as.character(CvGCpGislandsdf$seqnames)
head(CvGCpGislandsdf)
CvGCpGislandschrdf <- CvGCpGislandsdf[grep("LG",CvGCpGislandsdf$seqnames),]
head(CvGCpGislandschrdf)
CvGCpGislandschrdf <- CvGCpGislandschrdf[order(CvGCpGislandschrdf[,11]), ]
head(CvGCpGislandschrdf,10)
rm(CvGCpGislands)
rm(CvGCpGislandsdf)

```


```{r, CvL CpG island annotations}

ol1 <- findOverlaps(dmrseqCvL,my_CpGislands)
ol1
CvLCpGislands <- dmrseqCvL[queryHits(ol1)]
my_CpGislandsol1 <- my_CpGislands[subjectHits(ol1)]
head(CvLCpGislands)
my_CpGislandsol1

CvLCpGislandsdf <- as.data.frame(CvLCpGislands,row.names = 1:nrow(as.data.frame(ranges(CvLCpGislands))))
str(CvLCpGislandsdf)
my_CpGislandsol1 <- as.data.frame(my_CpGislandsol1,row.names = 1:nrow(as.data.frame(ranges(my_CpGislandsol1))))
head(my_CpGislandsol1)
CvLCpGislandsdf <- cbind(CvLCpGislandsdf,my_CpGislandsol1[c("source","ID")])
CvLCpGislandsdf$seqnames <- as.character(CvLCpGislandsdf$seqnames)
head(CvLCpGislandsdf)
CvLCpGislandschrdf <- CvLCpGislandsdf[grep("LG",CvLCpGislandsdf$seqnames),]
head(CvLCpGislandschrdf)
CvLCpGislandschrdf <- CvLCpGislandschrdf[order(CvLCpGislandschrdf[,11]), ]
head(CvLCpGislandschrdf,10)
rm(CvLCpGislands)
rm(CvLCpGislandsdf)

```


```{r, GvL CpG island annotations}

ol1 <- findOverlaps(dmrseqGvL,my_CpGislands)
ol1
GvLCpGislands <- dmrseqGvL[queryHits(ol1)]
my_CpGislandsol1 <- my_CpGislands[subjectHits(ol1)]
head(GvLCpGislands)
my_CpGislandsol1

GvLCpGislandsdf <- as.data.frame(GvLCpGislands,row.names = 1:nrow(as.data.frame(ranges(GvLCpGislands))))
str(GvLCpGislands)
my_CpGislandsol1 <- as.data.frame(my_CpGislandsol1,row.names = 1:nrow(as.data.frame(ranges(my_CpGislandsol1))))
head(my_CpGislandsol1)
GvLCpGislandsdf <- cbind(GvLCpGislandsdf,my_CpGislandsol1[c("source","ID")])
GvLCpGislandsdf$seqnames <- as.character(GvLCpGislandsdf$seqnames)
head(GvLCpGislandsdf)
GvLCpGislandschrdf <- GvLCpGislandsdf[grep("LG",GvLCpGislandsdf$seqnames),]
head(GvLCpGislandschrdf)
GvLCpGislandschrdf <- GvLCpGislandschrdf[order(GvLCpGislandschrdf[,11]), ]
head(GvLCpGislandschrdf,10)
rm(GvLCpGislands)
rm(GvLCpGislandsdf)

```

## dmrs annotated to promoter regions

```{r, CvG promoter annotations}

ol1 <- findOverlaps(dmrseqCvG,my_Promoters)
ol1
CvGpromoters <- dmrseqCvG[queryHits(ol1)]
myPromotersol1 <- my_Promoters[subjectHits(ol1)]
head(CvGpromoters)
myPromotersol1

CvGpromotersdf <- as.data.frame(CvGpromoters,row.names = 1:nrow(as.data.frame(ranges(CvGpromoters))))
str(CvGpromotersdf)
myPromotersoldf1 <- as.data.frame(myPromotersol1,row.names = 1:nrow(as.data.frame(ranges(myPromotersol1))))
head(myPromotersoldf1)
CvGpromotersdf <- cbind(CvGpromotersdf,myPromotersoldf1[c("V5","V6","V7")])
CvGpromotersdf$seqnames <- as.character(CvGpromotersdf$seqnames)
head(CvGpromotersdf)
CvGpromoterschrdf <- CvGpromotersdf[grep("LG",CvGpromotersdf$seqnames),]
head(CvGpromoterschrdf)
CvGpromoterschrdf <- CvGpromoterschrdf[order(CvGpromoterschrdf[,11]), ]
colnames(CvGpromoterschrdf) <- c("location","start","end","width","strand","L","area", "beta","stat", "pvalue","qvalue","index.start", "index.width","L2", "gene_id","transcript_id", "gene_name")
CvGpromoterschrdf <- dplyr::select(CvGpromoterschrdf,location,width,beta,pvalue,qvalue,gene_name)
head(CvGpromoterschrdf,10)
rm(CvGpromoters)
rm(CvGpromotersdf)

```


```{r, CvL promoter annotations}

ol1 <- findOverlaps(dmrseqCvL,my_Promoters)
ol1
CvLpromoters <- dmrseqCvL[queryHits(ol1)]
myPromotersol1 <- my_Promoters[subjectHits(ol1)]
head(CvLpromoters)
myPromotersol1

CvLpromotersdf <- as.data.frame(CvLpromoters,row.names = 1:nrow(as.data.frame(ranges(CvLpromoters))))
str(CvLpromotersdf)
myPromotersoldf1 <- as.data.frame(myPromotersol1,row.names = 1:nrow(as.data.frame(ranges(myPromotersol1))))
head(myPromotersoldf1)
CvLpromotersdf <- cbind(CvLpromotersdf,myPromotersoldf1[c("V5","V6","V7")])
CvLpromotersdf$seqnames <- as.character(CvLpromotersdf$seqnames)
head(CvLpromotersdf)
CvLpromoterschrdf <- CvLpromotersdf[grep("LG",CvLpromotersdf$seqnames),]
head(CvLpromoterschrdf)
CvLpromoterschrdf <- CvLpromoterschrdf[order(CvLpromoterschrdf[,11]), ]
colnames(CvLpromoterschrdf) <- c("location","start","end","width","strand","L","area", "beta","stat", "pvalue","qvalue","index.start", "index.width","L2", "gene_id","transcript_id", "gene_name")
CvLpromoterschrdf <- dplyr::select(CvLpromoterschrdf,location,width,beta,pvalue,qvalue,gene_name)
head(CvLpromoterschrdf,10)
rm(CvLpromoters)
rm(CvLpromotersdf)

```


```{r, GvL promoter annotations}

ol1 <- findOverlaps(dmrseqGvL,my_Promoters)
ol1
GvLpromoters <- dmrseqGvL[queryHits(ol1)]
myPromotersol1 <- my_Promoters[subjectHits(ol1)]
head(GvLpromoters)
myPromotersol1

GvLpromotersdf <- as.data.frame(GvLpromoters,row.names = 1:nrow(as.data.frame(ranges(GvLpromoters))))
str(GvLpromotersdf)
myPromotersoldf1 <- as.data.frame(myPromotersol1,row.names = 1:nrow(as.data.frame(ranges(myPromotersol1))))
head(myPromotersoldf1)
GvLpromotersdf <- cbind(GvLpromotersdf,myPromotersoldf1[c("V5","V6","V7")])
GvLpromotersdf$seqnames <- as.character(GvLpromotersdf$seqnames)
head(GvLpromotersdf)
GvLpromoterschrdf <- GvLpromotersdf[grep("LG",GvLpromotersdf$seqnames),]
head(GvLpromoterschrdf)
GvLpromoterschrdf <- GvLpromoterschrdf[order(GvLpromoterschrdf[,11]), ]
colnames(GvLpromoterschrdf) <- c("location","start","end","width","strand","L","area", "beta","stat", "pvalue","qvalue","index.start", "index.width","L2", "gene_id","transcript_id", "gene_name")
GvLpromoterschrdf <- dplyr::select(GvLpromoterschrdf,location,width,beta,pvalue,qvalue,gene_name)
head(GvLpromoterschrdf,20)
rm(GvLpromoters)
rm(GvLpromotersdf)

```

```{r, table 3}

CvGpromoterschrdf
CvGpromoterschrdf <- head(CvGpromoterschrdf,10)
CvGpromoterschrdf
CvLpromoterschrdf <- head(CvLpromoterschrdf,10)
CvLpromoterschrdf
GvLpromoterschrdf <-head(GvLpromoterschrdf,10)
GvLpromoterschrdf
LightComp <- c('CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG','CvG')
LightComp
CvGanodmrs <- cbind(LightComp,CvGpromoterschrdf)
saveRDS(object=CvGanodmrs,file="/mnt/data/aaron/projects/guppy-methylation/Thesis/figures/Tables/CvGanodmrs.rds")
```


```{r, table 4}

LightComp <- c('CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL','CvL')
CvLanodmrs <- cbind(LightComp,CvLpromoterschrdf)
saveRDS(object=CvLanodmrs,file="/mnt/data/aaron/projects/guppy-methylation/Thesis/figures/Tables/CvLanodmrs.rds")
```


```{r, table 5}

LightComp <- c('GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL','GvL')
GvLanodmrs <- cbind(LightComp,GvLpromoterschrdf)
saveRDS(object=GvLanodmrs,file="/mnt/data/aaron/projects/guppy-methylation/Thesis/figures/Tables/GvLanodmrs.rds")
```

## Create RCircos plot for CvG comparison

```{r, Gene label data CvG}

CvGgeneschrdfdmrs <- head(CvGgeneschrdf,75)
head(CvGgeneschrdfdmrs)
Chr.Label.Data <- dplyr::select(CvGgeneschrdfdmrs,seqnames,start,end,gene_name)
colnames(Chr.Label.Data) <- c("Chromosome","chromStart","chromEnd","GeneName")
head(Chr.Label.Data)
Chr.Label.Data <- Chr.Label.Data[-which(Chr.Label.Data$GeneName == "" | is.na(Chr.Label.Data$GeneName)), ]
head(Chr.Label.Data,10)

```


```{r, Input RCircos histogram values CvG}

head(CvGgeneschrdf)
RCircos.histogramV <- dplyr::select(CvGgeneschrdf,seqnames,start,end,pval)
head(RCircos.histogramV)
RCircos.histogramV$pval <- log(RCircos.histogramV$pval)
head(RCircos.histogramV)
colnames(RCircos.histogramV) <- c("Chromosome","chromStart","chromEnd","Data")
head(RCircos.histogramV)

```




```{r, Input RCircos heatmap values CvG}

head(CvGgeneschrdf)
RCircos.heatmapV <- dplyr::select(CvGgeneschrdf,seqnames,start,end,gene_name,beta)
head(RCircos.heatmapV)
colnames(RCircos.heatmapV) <- c("Chromosome","chromStart","chromEnd","GeneName","CvG")
head(RCircos.heatmapV)

```




```{r, ideogram data CvG}

data(UCSC.Mouse.GRCm38.CytoBandIdeogram)
str(UCSC.Mouse.GRCm38.CytoBandIdeogram)
chrlen <- read.table("/mnt/data/aaron/projects/guppy-methylation/chrlen.tsv")
str(chrlen)
segs <- apply(X = chrlen,MARGIN = 1,FUN = function(l){
chr <- l[1]
len <- l[2]
seg <- seq(1,len,1000000)
df <- data.frame(chr,as.integer(seg),as.integer(seg+999999))
df$band <- paste("x",1:nrow(df),sep = "")
stain <- c("gneg","gpos")
df$stain <- as.vector(matrix(data = stain,nrow = nrow(df),ncol = 1))
return(df)
})
segs <- do.call(rbind, segs)
colnames(segs) <- c("Chromosome","ChromStart","ChromEnd","Band","Stain")
str(segs)
head(segs)

```


```{r, Setup RCircos core components and Initialize Graphic Device CvG}

RCircos.Set.Core.Components(cyto.info = segs,tracks.inside = 4,chr.exclude = NULL,tracks.outside = 0)
rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
rcircos.params$hist.width <- 30
rcircos.params$heatmap.width <- 30
rcircos.params$text.size <- 0.75
RCircos.Reset.Plot.Parameters(new.params = rcircos.params)
#out.file <- "RCircosGuppyMethylome.pdf";
#pdf(file=out.file, height=8, width=8, compress=TRUE)

```




```{r, Create RCircos plot CvG, fig.height=10,fig.width=10}

#Setup RCircos plot
plot.new()
RCircos.Set.Plot.Area()
#Include ideogram
RCircos.Chromosome.Ideogram.Plot()
data(RCircos.histogramV)
#Create Heatmap
data(RCircos.heatmapV)
RCircos.Heatmap.Plot(heatmap.data = head(RCircos.heatmapV,75), data.col = 5,track.num = 1,side = "in")
#Create histogram
RCircos.Histogram.Plot(head(RCircos.histogramV,75),data.col=4, track.num=2, side="in")
#Include gene names and create connectors to histogram
RCircos.Gene.Connector.Plot(genomic.data = Chr.Label.Data,track.num = 3, side = "in")
RCircos.Gene.Name.Plot(head(Chr.Label.Data,75), name.col=4,track.num=4, side="in")

```


## Create RCircos plot for CvL comparison

```{r, Gene label data CvL}

CvLgeneschrdfdmrs <- head(CvLgeneschrdf,75)
head(CvLgeneschrdfdmrs)
Chr.Label.Data <- dplyr::select(CvLgeneschrdfdmrs,seqnames,start,end,gene_name)
colnames(Chr.Label.Data) <- c("Chromosome","chromStart","chromEnd","GeneName")
head(Chr.Label.Data)
Chr.Label.Data <- Chr.Label.Data[-which(Chr.Label.Data$GeneName == "" | is.na(Chr.Label.Data$GeneName)), ]
head(Chr.Label.Data, 10)

```


```{r, Input RCircos histogram values CvL}

head(CvLgeneschrdf)
RCircos.histogramV <- dplyr::select(CvLgeneschrdf,seqnames,start,end,pval)
head(RCircos.histogramV)
RCircos.histogramV$pval <- log(RCircos.histogramV$pval)
head(RCircos.histogramV)
colnames(RCircos.histogramV) <- c("Chromosome","chromStart","chromEnd","Data")
head(RCircos.histogramV)

```




```{r, Input RCircos heatmap values CvL}

head(CvLgeneschrdf)
RCircos.heatmapV <- dplyr::select(CvLgeneschrdf,seqnames,start,end,gene_name,beta)
head(RCircos.heatmapV)
colnames(RCircos.heatmapV) <- c("Chromosome","chromStart","chromEnd","GeneName","CvG")
head(RCircos.heatmapV)

```




```{r, Setup RCircos core components and Initialize Graphic Device CvL}

RCircos.Set.Core.Components(cyto.info = segs,tracks.inside = 4,chr.exclude = NULL,tracks.outside = 0)
rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
rcircos.params$hist.width <- 30
rcircos.params$heatmap.width <- 30
rcircos.params$text.size <- 0.75
RCircos.Reset.Plot.Parameters(new.params = rcircos.params)
#out.file <- "RCircosGuppyMethylome.pdf";
#pdf(file=out.file, height=8, width=8, compress=TRUE)

```




```{r, Create RCircos plot CvL, fig.height=10,fig.width=10}

#Setup RCircos plot
plot.new()
RCircos.Set.Plot.Area()
#Include ideogram
RCircos.Chromosome.Ideogram.Plot()
data(RCircos.histogramV)
#Create Heatmap
data(RCircos.heatmapV)
RCircos.Heatmap.Plot(heatmap.data = head(RCircos.heatmapV,75), data.col = 5,track.num = 1,side = "in")
#Create histogram
RCircos.Histogram.Plot(head(RCircos.histogramV,75),data.col=4, track.num=2, side="in")
#Include gene names and create connectors to histogram
RCircos.Gene.Connector.Plot(genomic.data = Chr.Label.Data,track.num = 3, side = "in")
RCircos.Gene.Name.Plot(head(Chr.Label.Data,75), name.col=4,track.num=4, side="in")

```


## Create RCircos plot for GvL comparison

```{r, Gene label data GvL}

CvLgeneschrdfdmrs <- head(GvLgeneschrdf,75)
head(CvLgeneschrdfdmrs)
Chr.Label.Data <- dplyr::select(CvLgeneschrdfdmrs,seqnames,start,end,gene_name)
colnames(Chr.Label.Data) <- c("Chromosome","chromStart","chromEnd","GeneName")
head(Chr.Label.Data)
Chr.Label.Data <- Chr.Label.Data[-which(Chr.Label.Data$GeneName == "" | is.na(Chr.Label.Data$GeneName)), ]
head(Chr.Label.Data,10)

```



```{r, Input RCircos histogram values GvL}

head(GvLgeneschrdf)
RCircos.histogramV <- dplyr::select(GvLgeneschrdf,seqnames,start,end,pval)
head(RCircos.histogramV)
RCircos.histogramV$pval <- log(RCircos.histogramV$pval)
head(RCircos.histogramV)
colnames(RCircos.histogramV) <- c("Chromosome","chromStart","chromEnd","Data")
head(RCircos.histogramV)

```




```{r, Input RCircos heatmap values GvL}

head(GvLgeneschrdf)
RCircos.heatmapV <- dplyr::select(GvLgeneschrdf,seqnames,start,end,gene_name,beta)
head(RCircos.heatmapV)
colnames(RCircos.heatmapV) <- c("Chromosome","chromStart","chromEnd","GeneName","CvG")
head(RCircos.heatmapV)

```



```{r, Setup RCircos core components and Initialize Graphic Device GvL}

RCircos.Set.Core.Components(cyto.info = segs,tracks.inside = 4,chr.exclude = NULL,tracks.outside = 0)
rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
rcircos.params$hist.width <- 30
rcircos.params$heatmap.width <- 30
rcircos.params$text.size <- 0.75
RCircos.Reset.Plot.Parameters(new.params = rcircos.params)
#out.file <- "RCircosGuppyMethylome.pdf";
#pdf(file=out.file, height=8, width=8, compress=TRUE)

```



```{r, Create RCircos plot GvL, fig.height=10,fig.width=10}

#Setup RCircos plot
plot.new()
RCircos.Set.Plot.Area()
#Include ideogram
RCircos.Chromosome.Ideogram.Plot()
data(RCircos.histogramV)
#Create Heatmap
data(RCircos.heatmapV)
RCircos.Heatmap.Plot(heatmap.data = head(RCircos.heatmapV,75), data.col = 5,track.num = 1,side = "in")
#Create histogram
RCircos.Histogram.Plot(head(RCircos.histogramV,75),data.col=4, track.num=2, side="in")
#Include gene names and create connectors to histogram
RCircos.Gene.Connector.Plot(genomic.data = Chr.Label.Data,track.num = 3, side = "in")
RCircos.Gene.Name.Plot(head(Chr.Label.Data,75), name.col=4,track.num=4, side="in")

```


## Read in the methylation data from Biscuit

```{r, Read in percentage methylation for each sample}

ClearR1 <- read.table("meth_data/Clear2F-R1.bam.vcf.gz.cg.bed.gz")
ClearR2 <- read.table("meth_data/Clear2F-R2.bam.vcf.gz.cg.bed.gz")
ClearR3 <- read.table("meth_data/Clear2F-R3.bam.vcf.gz.cg.bed.gz")
Foundation <- read.table("meth_data/Foundation.bam.vcf.gz.cg.bed.gz")
GreenR1 <- read.table("meth_data/Green3F-R1.bam.vcf.gz.cg.bed.gz")
GreenR2 <- read.table("meth_data/Green3F-R2.bam.vcf.gz.cg.bed.gz")
GreenR3 <- read.table("meth_data/Green3F-R3.bam.vcf.gz.cg.bed.gz")
LilacR1 <- read.table("meth_data/Lilac4F-R1.bam.vcf.gz.cg.bed.gz")
LilacR2 <- read.table("meth_data/Lilac4F-R2.bam.vcf.gz.cg.bed.gz")
LilacR3 <- read.table("meth_data/Lilac4F-R3.bam.vcf.gz.cg.bed.gz")

```

## Make some plots to investigate CpG methylation in DMRs

```{r, subset samples for CvG dmrs,fig.height=8,fig.width=8}

library(vioplot)
library(beeswarm)

dmrplots <- function(dmr){

  dmrseqname <- dmr[1]
  dmrstart <- dmr[2]
  dmrend <- dmr[3]
  gene_name <- dmr[11]
  
  Foundationsub <- Foundation[which( Foundation$V1==dmrseqname & Foundation$V2>dmrstart & Foundation$V3<dmrend ),4]
  ClearR1sub <- ClearR1[which( ClearR1$V1==dmrseqname & ClearR1$V2>dmrstart & ClearR1$V3<dmrend ),4]
  ClearR2sub <- ClearR2[which( ClearR2$V1==dmrseqname & ClearR2$V2>dmrstart & ClearR2$V3<dmrend ),4]
  ClearR3sub <- ClearR3[which( ClearR3$V1==dmrseqname & ClearR3$V2>dmrstart & ClearR3$V3<dmrend ),4]
  GreenR1sub <- GreenR1[which( GreenR1$V1==dmrseqname & GreenR1$V2>dmrstart & GreenR1$V3<dmrend ),4]
  GreenR2sub <- GreenR2[which( GreenR2$V1==dmrseqname & GreenR2$V2>dmrstart & GreenR2$V3<dmrend ),4]
  GreenR3sub <- GreenR3[which( GreenR3$V1==dmrseqname & GreenR3$V2>dmrstart & GreenR3$V3<dmrend ),4]
  LilacR1sub <- LilacR1[which( LilacR1$V1==dmrseqname & LilacR1$V2>dmrstart & LilacR1$V3<dmrend ),4]
  LilacR2sub <- LilacR2[which( LilacR2$V1==dmrseqname & LilacR2$V2>dmrstart & LilacR2$V3<dmrend ),4]
  LilacR3sub <- LilacR3[which( LilacR3$V1==dmrseqname & LilacR3$V2>dmrstart & LilacR3$V3<dmrend ),4]
  
  ml <- list("Foundation"=Foundationsub,
             "ClearR1"=ClearR1sub,
             "ClearR2"=ClearR2sub,
             "ClearR3"=ClearR3sub,
             "GreenR1"=GreenR1sub,
             "GreenR2"=GreenR2sub,
             "GreenR3"=GreenR3sub,
             "LilacR1"=LilacR1sub,
             "LilacR2"=LilacR2sub,
             "LilacR3"=LilacR3sub)
  HEADER = paste(dmrseqname,dmrstart,dmrend,gene_name)
  par(mar=c(5,10,3,1))
  vioplot(ml, main = HEADER, xlab = "% Methylation",  horizontal=TRUE,las=1)
  beeswarm(ml, main = HEADER,  xlab = "% Methylation", horizontal = TRUE, las=1,pch=19,cex=0.5)
  mlm <- unlist(lapply(ml,mean))
  barplot( mlm, main = HEADER, xlab = "% Methylation", horiz=TRUE,las=1,xlim=c(min(mlm)-0.01,max(mlm)+0.01))
}

# test that one works
# dmr <- CvGgeneschrdf[1,,drop=TRUE]
# dmrplots(dmr)

# with lapply
lapply(X=1:10, function(i) {
  dmr <- CvGpromoterschrdf[i,,drop=TRUE]
  dmrplots(dmr)
})


```



```{r, subset samples for CvL dmrs,fig.height=8,fig.width=8}

library(vioplot)
library(beeswarm)

dmrplots <- function(dmr){

  dmrseqname <- dmr[1]
  dmrstart <- dmr[2]
  dmrend <- dmr[3]
  gene_name <- dmr[17]
  
  Foundationsub <- Foundation[which( Foundation$V1==dmrseqname & Foundation$V2>dmrstart & Foundation$V3<dmrend ),4]
  ClearR1sub <- ClearR1[which( ClearR1$V1==dmrseqname & ClearR1$V2>dmrstart & ClearR1$V3<dmrend ),4]
  ClearR2sub <- ClearR2[which( ClearR2$V1==dmrseqname & ClearR2$V2>dmrstart & ClearR2$V3<dmrend ),4]
  ClearR3sub <- ClearR3[which( ClearR3$V1==dmrseqname & ClearR3$V2>dmrstart & ClearR3$V3<dmrend ),4]
  GreenR1sub <- GreenR1[which( GreenR1$V1==dmrseqname & GreenR1$V2>dmrstart & GreenR1$V3<dmrend ),4]
  GreenR2sub <- GreenR2[which( GreenR2$V1==dmrseqname & GreenR2$V2>dmrstart & GreenR2$V3<dmrend ),4]
  GreenR3sub <- GreenR3[which( GreenR3$V1==dmrseqname & GreenR3$V2>dmrstart & GreenR3$V3<dmrend ),4]
  LilacR1sub <- LilacR1[which( LilacR1$V1==dmrseqname & LilacR1$V2>dmrstart & LilacR1$V3<dmrend ),4]
  LilacR2sub <- LilacR2[which( LilacR2$V1==dmrseqname & LilacR2$V2>dmrstart & LilacR2$V3<dmrend ),4]
  LilacR3sub <- LilacR3[which( LilacR3$V1==dmrseqname & LilacR3$V2>dmrstart & LilacR3$V3<dmrend ),4]
  
  ml <- list("Foundation"=Foundationsub,
             "ClearR1"=ClearR1sub,
             "ClearR2"=ClearR2sub,
             "ClearR3"=ClearR3sub,
             "GreenR1"=GreenR1sub,
             "GreenR2"=GreenR2sub,
             "GreenR3"=GreenR3sub,
             "LilacR1"=LilacR1sub,
             "LilacR2"=LilacR2sub,
             "LilacR3"=LilacR3sub)
  HEADER = paste(dmrseqname,dmrstart,dmrend,gene_name)
  par(mar=c(5,10,3,1))
  vioplot(ml, main = HEADER, xlab = "% Methylation",  horizontal=TRUE,las=1)
  beeswarm(ml, main = HEADER,  xlab = "% Methylation", horizontal = TRUE, las=1,pch=19,cex=0.5)
  mlm <- unlist(lapply(ml,mean))
  barplot( mlm, main = HEADER, xlab = "% Methylation", horiz=TRUE,las=1,xlim=c(min(mlm)-0.01,max(mlm)+0.01))
}

# test that one works
# dmr <- CvLgeneschrdf[1,,drop=TRUE]
# dmrplots(dmr)

# with lapply
lapply(X=1:10, function(i) {
  dmr <- CvLpromoterschrdf[i,,drop=TRUE]
  dmrplots(dmr)
})


```




```{r, subset samples for GvL dmrs,fig.height=8,fig.width=8}

library(vioplot)
library(beeswarm)

dmrplots <- function(dmr){

  dmrseqname <- dmr[1]
  dmrstart <- dmr[2]
  dmrend <- dmr[3]
  gene_name <- dmr[17]
  
  Foundationsub <- Foundation[which( Foundation$V1==dmrseqname & Foundation$V2>dmrstart & Foundation$V3<dmrend ),4]
  ClearR1sub <- ClearR1[which( ClearR1$V1==dmrseqname & ClearR1$V2>dmrstart & ClearR1$V3<dmrend ),4]
  ClearR2sub <- ClearR2[which( ClearR2$V1==dmrseqname & ClearR2$V2>dmrstart & ClearR2$V3<dmrend ),4]
  ClearR3sub <- ClearR3[which( ClearR3$V1==dmrseqname & ClearR3$V2>dmrstart & ClearR3$V3<dmrend ),4]
  GreenR1sub <- GreenR1[which( GreenR1$V1==dmrseqname & GreenR1$V2>dmrstart & GreenR1$V3<dmrend ),4]
  GreenR2sub <- GreenR2[which( GreenR2$V1==dmrseqname & GreenR2$V2>dmrstart & GreenR2$V3<dmrend ),4]
  GreenR3sub <- GreenR3[which( GreenR3$V1==dmrseqname & GreenR3$V2>dmrstart & GreenR3$V3<dmrend ),4]
  LilacR1sub <- LilacR1[which( LilacR1$V1==dmrseqname & LilacR1$V2>dmrstart & LilacR1$V3<dmrend ),4]
  LilacR2sub <- LilacR2[which( LilacR2$V1==dmrseqname & LilacR2$V2>dmrstart & LilacR2$V3<dmrend ),4]
  LilacR3sub <- LilacR3[which( LilacR3$V1==dmrseqname & LilacR3$V2>dmrstart & LilacR3$V3<dmrend ),4]
  
  ml <- list("Foundation"=Foundationsub,
             "ClearR1"=ClearR1sub,
             "ClearR2"=ClearR2sub,
             "ClearR3"=ClearR3sub,
             "GreenR1"=GreenR1sub,
             "GreenR2"=GreenR2sub,
             "GreenR3"=GreenR3sub,
             "LilacR1"=LilacR1sub,
             "LilacR2"=LilacR2sub,
             "LilacR3"=LilacR3sub)
  HEADER = paste(dmrseqname,dmrstart,dmrend,gene_name)
  par(mar=c(5,10,3,1))
  vioplot(ml, main = HEADER, xlab = "% Methylation",  horizontal=TRUE,las=1)
  beeswarm(ml, main = HEADER,  xlab = "% Methylation", horizontal = TRUE, las=1,pch=19,cex=0.5)
  mlm <- unlist(lapply(ml,mean))
  barplot( mlm, main = HEADER, xlab = "% Methylation", horiz=TRUE,las=1,xlim=c(min(mlm)-0.01,max(mlm)+0.01))
}

# test that one works
# dmr <- GvLgeneschrdf[1,,drop=TRUE]
# dmrplots(dmr)

# with lapply
lapply(X=1:10, function(i) {
  dmr <- GvLpromoterschrdf[i,,drop=TRUE]
  dmrplots(dmr)
})


```


## Session Information

For reproducibility

```{r,sessioninfo}

sessionInfo()

```